src/core/qgstransaction.cpp

```diff
@@ -189,3 +194,54 @@ void QgsTransaction::setLayerTransactionIds( QgsTransaction *transaction )
     }
   }
 }
+
+QString QgsTransaction::createSavepoint( QString &error SIP_OUT )
+{
+  if ( !mTransactionActive )
+    return QString();
+
+  if ( !mLastSavePointIsDirty )
+    return mSavepoints.top();
+
+  const QString name( QUuid::createUuid().toString() );
+
+  if ( !executeSql( "SAVEPOINT \"" + name + "\"", error ) )
```

Comment:
`QStringLiteral().arg()` may be use here with something like `QgsExpression::quotedColumnRef` or `QgsExpression::quotedString` to manage quotes. What do you think?

Reply:
(elsewhere in the code too)

Reply:
Is there a good reason for that ? Because I'm not sure it improves clarity or robustness.

Reply:
One minor advantage is less escaping for better readability. Anyway, `QStringLieteral` should be used in either case.

Reply:
Ok for consistency, ok robustness, but readability:
```
QStringLiteral("SAVEPOINT %1").arg( QgsExpression::quotedString( name ) )
```

Reply:
I just like template based solutions over concatenation ;)